### PR TITLE
build(tablebench): use node16 module resolution

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automatic PR to switch to node16 module resolution

Also removes several unintentionally copied tsconfig settings.

...and fixes use of internal ISharedTree -> public ITree

### CODE REVIEW

File: examples/benchmarks/tablebench/src/test/tsconfig.json [hunk 1]
```diff
@@ -1,14 +1,9 @@
...
+	"extends": "../../../../../common/build/build-common/tsconfig.test.node16.json",
 	"compilerOptions": {
-		"lib": ["ES2019", "ES2018.Promise", "DOM", "DOM.Iterable"],
-		"noImplicitAny": true,
-		"noUnusedLocals": false,
-		"noImplicitOverride": true,
 		"rootDir": "./",
-		"outDir": "../../dist/test",
-		"types": ["node", "mocha"],
-		"resolveJsonModule": true,
```
Comment (jason-ha) [inline examples/benchmarks/tablebench/src/test/tsconfig.json:11]: Can the deltas be noted in the description?
Things I see:
`"resolveJsonModule": true,`  - removed; if viable cool
`"noImplicitAny": true,` - removed and should be restored
`"noImplicitOverride": true,` - removed and should be restored
lib changes - okay to update?

File: examples/benchmarks/tablebench/src/test/table.bench.spec.ts [hunk 2]
```diff
@@ -3,12 +3,12 @@
  * Licensed under the MIT License.
  */
 
-import { NodeFromSchema, SharedTree, type ISharedTree } from "@fluidframework/tree";
+import { NodeFromSchema, SharedTree, type ITree } from "@fluidframework/tree";
```
Comment (jason-ha) [inline examples/benchmarks/tablebench/src/test/table.bench.spec.ts:6]: note the type change in description

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: examples/benchmarks/tablebench/src/test/tsconfig.json (hunk 1 above)
Comment (DLehenbauer) [reply examples/benchmarks/tablebench/src/test/tsconfig.json:11]: @jason-ha - I'm the author of TableBench.  All of these settings were unintentionally copy/pasted from another example package.

The only one that I added was 'resolveJsonModule' which is no longer necessary (I used to import a JSON file for the benchmark, now I synthetically generate statistically similar test data.)

Comment (jason-ha) [reply examples/benchmarks/tablebench/src/test/tsconfig.json:11]: Those two `noImplicit*` settings are ones that we *should* have everywhere. They are stricter, so, I would _vote_ for keeping them.

Comment (jason-ha) [reply examples/benchmarks/tablebench/src/test/tsconfig.json:11]: And you did :)

**Review by jason-ha — APPROVED** [review]

**Pull request merged**